packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -268,35 +269,11 @@ const DropdownMenuRootContent = React.forwardRef<
       portalled={portalled}
       onCloseAutoFocus={(event) => {
         props.onCloseAutoFocus?.(event);
-
         if (!event.defaultPrevented) {
-          if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
+          if (context.modal) context.triggerRef.current?.focus();
```

Comment:
I realised I could delete all of the `onInteractOutside` code if I just checked for modality here. If it's non-modal then we just want the browser to focus whatever they clicked which will automatically happen because we `event.preventDefault()` our `onCloseAutoFocus` below.

Reply:
> Is it an issue that it focuses?

My concern is how this affects the next tab position and interferes with text selection / cursor position inside inputs.

Reply:
Ah, yes, for some reason in my mind I expected that the thing I right clicked on **wouldn't** be focused because it was inert when I clicked it. But I can see the content in the context menu relates to the the thing they right clicked so it should probably be focused!

Okay, I'll add some logic back in to handle this.

Reply:
Hmmm, this is why I found it weird:

https://user-images.githubusercontent.com/175330/149388758-dffaf212-622e-494e-9fff-5ec383bd96ac.mp4

When it's modal and you **click** outside, the trigger focuses. But when it's modal and you **right click** outside, the textarea focuses. Why the difference in expectation here?

Reply:
> Why the difference in expectation here?

This was modelled after native behaviour if I recall.

https://user-images.githubusercontent.com/11708259/149390858-9363d32d-a9df-472c-a25d-9d0bf56a1b73.mp4

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

Reply:
Which app is that? MacOS Notes doesn't behave like ours on left click let alone right click so I'm not sure what we should be following:

https://user-images.githubusercontent.com/175330/149395660-a254f1c3-d65b-450b-8877-d205754f31bc.mp4

When I left click the menubar, it focuses the textarea 😅

Reply:
Okay lads, I've pushed code that makes it behave like the [first video here](https://github.com/radix-ui/primitives/pull/1082#discussion_r784226074). I'm confused but if that's how it was before and for good reason then l'm happy to stay confused 😅

Reply:
> Which app is that?

Yeah MacOS Notes

Native select is probably a clearer example:

https://user-images.githubusercontent.com/11708259/149396585-a1c7257b-673e-4bf1-a5a8-f24fca62c922.mp4

I agree with the point about consistency, but it was definitely something we discussed at the time and was an intentional decision. As a user I do find it more helpful than not.

Reply:
Ah, that `select` is a great example, cheers 👍

Code reflects that now 💛

Reply:
Just adding my 2 cents here that yes, this behaviour was on purpose as I had shown the same types of native examples to @andy-hook back then, right-click is treated differently.